src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue`1.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+
+namespace Roslyn.Utilities
+{
+    /// <inheritdoc cref="AsyncBatchingWorkQueue{TItem, TResult}"/>
+    internal class AsyncBatchingWorkQueue<TItem> : AsyncBatchingWorkQueue<TItem, VoidResult>
+    {
+        public AsyncBatchingWorkQueue(
+            TimeSpan delay,
+            Func<ImmutableArray<TItem>, CancellationToken, Task> processBatchAsync,
+            IAsynchronousOperationListener asyncListener,
+            CancellationToken cancellationToken)
+            : this(delay,
+                   processBatchAsync,
+                   equalityComparer: null,
+                   asyncListener,
+                   cancellationToken)
+        {
+        }
+
+        public AsyncBatchingWorkQueue(
+            TimeSpan delay,
+            Func<ImmutableArray<TItem>, CancellationToken, Task> processBatchAsync,
+            IEqualityComparer<TItem>? equalityComparer,
+            IAsynchronousOperationListener asyncListener,
+            CancellationToken cancellationToken)
+            : base(delay, Convert(processBatchAsync), equalityComparer, asyncListener, cancellationToken)
+        {
+        }
+
+        private static Func<ImmutableArray<TItem>, CancellationToken, Task<VoidResult>> Convert(Func<ImmutableArray<TItem>, CancellationToken, Task> processBatchAsync)
+            => async (items, ct) =>
+            {
+                await processBatchAsync(items, ct).ConfigureAwait(false);
+                return default;
```

Comment:
💡  The default state machine for `Task<VoidResult>` doesn't know how to cache a `default(VoidResult)`. However, since the result of calling `_processBatchAsync` is always immediately awaited in a simple matter, this delegate can be changed to return `ValueTask<VoidResult>` which does avoid the `Task<VoidResult>` allocation if the processing returns synchronously.

Reply:
done.